### ISSUE TITLE
file SCHEMA of IFC2X2_FINAL is not supported?

### ISSUE THREAD

**Issue by wssbygone** [IfcOpenShell/IfcOpenShell#560]
hello,  I've build the latest code and generated some exectuable binary, I wonder if the file schema **IFC2X2_FINAL** is not supported now?   Does the code is not backward compatibility?

it's the model below
[Munkerud.zip](https://github.com/IfcOpenShell/IfcOpenShell/files/2952337/Munkerud.zip)

thanks

**Comment by aothms** [comment]
No, there is no full backwards compatibility between IFC schemas. Earlier versions of IfcOpenShell accepted population files with a different schema than selected during compilation. Newer versions don't because of all sorts of subtle errors. For example I think between 2x2 and 2x3 the positioning between profiles was changed. The documentation for 2x2 is not available online so we can never be sure. Is there a real need supporting this?

**Comment by wssbygone** [comment]
Yes, Thanks! We really need to support IFC2X2_FINAL, because  there are some models with the file schema  IFC2X2_FINAL, we can parse these models until we replaced IfcGeomServer with newer version recently, although the newer version resolve some other issues. Could you tell me how to  change the selection during compilation to support IFC2X2_FINAL or some other solutions?

Thanks
Daniel

**Comment by aothms** [comment]
Another option is to simply remove the file schema check:

In master: https://github.com/IfcOpenShell/IfcOpenShell/blob/master/src/ifcparse/IfcParse.cpp#L1270

In v0.6 you'll need to select a schema: https://github.com/IfcOpenShell/IfcOpenShell/blame/cb535b4ba8679f8f157587cced0a52ed054fadc1/src/ifcparse/IfcParse.cpp#L1343

That's probably simplest.

**Comment by aothms** [comment]
In case it enables you to stay in sync more easily, feel free to send a Pull Request to implement something like this behind a command line arg like `--no-schema-check` or similar.

**Comment by wssbygone** [comment]
Ok, I'll modify the schema check later.  And I find some differences, this is the result(IfcOpenShell+WebGL) as shown below:
[comparison picture](https://github.com/wssbygone/mydepot/blob/master/new.old.ifc.png)

the model doesn't carry material infomation, so the parsing result of color arrary should be empty,
but now the latest version(master branch code), the color array is [0,0,0,1],  what adjustments have been made to this recently?

**Comment by aothms** [comment]
>  what adjustments have been made to this recently?

This probably https://github.com/IfcOpenShell/IfcOpenShell/pull/472 but pure black should not be the aim. Let me know if you find out more.

**Comment by wssbygone** [comment]
You can look at this image：
![new old ifc](https://user-images.githubusercontent.com/38974768/54605583-922be700-4a84-11e9-9ad4-e6ecc1b7d4b0.png)

**Comment by wssbygone** [comment]
I have read the code changes from  #472 , it  only adds the function **set_default_style_file** to customize the initilization color, this function is not called by any other places, I think the issue of pure black has nothing to do with #472,  atfer rollbacking the code about #490, the issue is fixed.

**Comment by Moult** [comment]
Closing as IFC2X2_FINAL is not an ISO schema and is unsupported.
